packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -845,7 +845,7 @@ class _BottomNavigationBarState extends State<BottomNavigationBar> with TickerPr
       );
     });
     _controllers[widget.currentIndex].value = 1.0;
-    _backgroundColor = widget.items[widget.currentIndex].backgroundColor;
+    _backgroundColor = widget.backgroundColor;
```

Comment:
This is going to need a widget test to verify this behavior.

You could try pumping a BottomNavigationBar with a background color, and give it a bunch of BottomNavigationBarItems that each have a color that is different from the main background color (and preferably also different from each other)

Then check in an `expect ()` that the background color is the background color for the BottomNavigationBar.